src/gradata/hooks/tool_finding_capture.py

```
@@ -0,0 +1,129 @@
+        tmp = FINDINGS_FILE.with_suffix(".tmp")
+        tmp.write_text(content, encoding="utf-8")
+        tmp.replace(FINDINGS_FILE)  # atomic on POSIX, near-atomic on Windows
+    except Exception:
+        pass
+
+
+def _extract_failed_files(output: str) -> list[str]:
+    """Extract file paths from test failure output."""
+    files = []
+    for line in output.splitlines():
+        line = line.strip()
```

Ruff (0.15.9)

[warning] `for` loop variable `line` overwritten by assignment target

(PLW2901)